auth/auth-server/src/server/handle_external_match.rs

```diff
@@ -640,12 +637,45 @@ impl Server {
                 .topic2(nonce)
                 .from_block(self.start_block_num);
 
-        let events = filter.query().await.map_err(AuthServerError::gas_sponsorship)?;
+        let events = filter.query_with_meta().await.map_err(AuthServerError::gas_sponsorship)?;
 
         // If no event was found, we assume that gas was not sponsored for this nonce.
         // This could be the case if the gas sponsor was underfunded or paused.
-        let amount_sponsored = events.last().map(|event| event.amount);
+        let amount_sponsored_with_tx =
+            events.last().map(|(event, meta)| (event.amount, meta.transaction_hash));
+
+        Ok(amount_sponsored_with_tx)
+    }
 
-        Ok(amount_sponsored)
+    /// Record the gas sponsorship rate limit & metrics for a given settled
+    /// match
+    async fn record_settled_match_sponsorship(
+        &self,
+        match_resp: &ExternalMatchResponse,
+        key: String,
+        request_id: String,
+    ) -> Result<(), AuthServerError> {
+        if match_resp.is_sponsored
+            && let Some((gas_cost, tx_hash)) =
+                self.get_sponsorship_amount_and_tx(&match_resp.match_bundle.settlement_tx).await?
+        {
+            // Convert wei to ether using format_ether, then parse to f64
+            let gas_cost_eth: f64 =
+                format_ether(gas_cost).parse().map_err(AuthServerError::custom)?;
+
+            let eth_price: f64 = self
+                .price_reporter_client
+                .get_eth_price()
+                .await
+                .map_err(AuthServerError::custom)?;
+
+            let gas_sponsorship_value = eth_price * gas_cost_eth;
+
+            self.record_gas_sponsorship_rate_limit(key, gas_sponsorship_value).await?;
+
```

Comment:
Seems like we can just invoke self.rate_limiter.record_gas_sponsorship() here, but maybe you have future need to abstract some logic away?

Reply:
fair point, needless indirection